### PR TITLE
[PIR & Inference] Add conv2dAddPass and conv2dAddActPass and conv2dAdd2ActPass

### PR DESCRIPTION
**Pull request by bukejiyu**
### PR types
Others 
### PR changes
Others
### Description
add conv2d_add pass and  conv2d_add unittest
add conv2d+add+act pass and unittest
add conv2d+add+add+act pass and unittest
fix constant_folding_pass
Pcard-71500

### CODE REVIEW

File: paddle/fluid/pir/transforms/fusion/conv2d_add_fuse_pass.cc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+#include "paddle/fluid/pir/transforms/fusion/conv2d_add_fuse_pass.h"
+#include "paddle/fluid/pir/transforms/transform_general_functions.h"
+
+#include "paddle/fluid/pir/drr/api/drr_pattern_base.h"
+#include "paddle/phi/core/ddim.h"
+#include "paddle/pir/pass/pass.h"
+
+namespace {
+
+class Conv2dAddFusePattern
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/fusion/conv2d_add_fuse_pass.cc:31]: 这个pattern直接放在conv2d_fuse那个pass里

Comment (yuanlehome) [reply paddle/fluid/pir/transforms/fusion/conv2d_add_fuse_pass.cc:31]: 不需要新增加一个conv2d_add_fuse_pass

File: paddle/fluid/inference/api/analysis_predictor.cc [hunk 2]
```diff
@@ -776,6 +777,7 @@ bool AnalysisPredictor::PrepareExecutor() {
       //----------------------------------------------------------------------------------------------//
       // Operator fusion pass
       pm_for_op_program.AddPass(::pir::CreateConv2dFusePass());
```
Comment (yuanlehome) [inline paddle/fluid/inference/api/analysis_predictor.cc:779]: 把conv2d_fuse_pass改名为conv2d_add_act_fuse_pass吧

File: paddle/fluid/pir/transforms/fusion/conv2d_fuse_pass.cc [hunk 3]
```diff
@@ -122,6 +120,135 @@ class Conv2dBnFusePattern
...
+    if (!conv2d_op) return false;
+
+    pir::OpResult conv2d_out = conv2d_op.out();
+    if (!conv2d_out.HasOneUse()) return false;
+
+    auto next_op_list = pir::GetUseOpsForOutput(add2_op, 0);
+    if (next_op_list.size() == 0) return false;
+    auto next_op = next_op_list[0];
+    std::string act_name = "";
+    if (next_op->dyn_cast<paddle::dialect::ReluOp>()) {
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/fusion/conv2d_fuse_pass.cc]: 仅判断使用，用isa，需要转换再用dyn_cast，下同

**Review by yuanlehome — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM
单测重命名

**Pull request merged**